### ISSUE TITLE
[Due for payment 2025-02-25]  Per diem - RHP shows offline view when deleted, fields are not grayed out when edited offline

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#54488]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 9.0.78-0
**Reproducible in staging?:** Y
**Reproducible in production?:** N
**If this was caught on HybridApp, is this reproducible on New Expensify Standalone?:** N/A
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Email or phone of affected tester (no customers):** applausetester+909608@applause.expensifail.com
**Issue reported by:** Applause - Internal Team

## Action Performed:
- Workspace has Per diem rates
1. Go to staging.new.expensify.com
2. Go to workspace settings > Per diem
3. Go offline
4. Click on any rate
5. Click Delete
6. Delete the rate
7. Do any edit offline for other rate

## Expected Result:
In Step 6, after deleting the rate, RHP will not show offline view

## Actual Result:
In Step 6, after deleting the rate, RHP shows offline view while dismissing

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Standalone
- [ ] Android: HybridApp
- [x] Android: mWeb Chrome
- [x] iOS: Standalone
- [ ] iOS: HybridApp
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

<details>
  <summary>Add any screenshot/video evidence</summary>

  
</details> 

https://github.com/user-attachments/assets/73eb2d31-8ab7-4625-8111-159bdb1cb574

<img width="1191" alt="Bug6701539_1734998541197!Screenshot_2024-12-24_at_08 00 06" src="https://github.com/user-attachments/assets/999cb3d1-0932-41cd-9eb3-da9f5da8f8e8" />

[Bug6701539_1734998541202!German_Per_Diem_-_Dependent_Tags___2_.csv](https://github.com/user-attachments/files/18234349/Bug6701539_1734998541202.German_Per_Diem_-_Dependent_Tags___2_.csv)

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

**Comment by jasperhuangg** [comment]
Yeah, lets please implement this as part of the original PR, it's definitely a regression that should be fixed. @parasharrajat @daledah

**Comment by jasperhuangg** [comment]
I'm going to remove the blocker label since this is pretty edge-casey, and it seems we need to sort out which pattern is intended here.

**Comment by huult** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.

RHP shows offline view when deleted, fields are not grayed out when edited offline

### What is the root cause of that problem?

When deleting a Per Diem rate in offline mode, the deleted rate is removed from Onyx, causing `selectedSubRate` to become `undefined`.

https://github.com/Expensify/App/blob/0ee0a123dac5b49f00da6c90619c03d5dabdd71f/src/pages/workspace/perDiem/WorkspacePerDiemDetailsPage.tsx#L41

So we will briefly show the offline page before closing the modal

https://github.com/Expensify/App/blob/0ee0a123dac5b49f00da6c90619c03d5dabdd71f/src/pages/workspace/perDiem/WorkspacePerDiemDetailsPage.tsx#L46C48-L46C63

https://github.com/Expensify/App/blob/0ee0a123dac5b49f00da6c90619c03d5dabdd71f/src/pages/workspace/perDiem/WorkspacePerDiemDetailsPage.tsx#L60

### What changes do you think we should make in order to solve the problem?

To resolve this issue, we should update the condition for displaying the offline page to ensure that it is not displayed unnecessarily after navigating back

https://github.com/Expensify/App/blob/0ee0a123dac5b49f00da6c90619c03d5dabdd71f/src/pages/workspace/perDiem/WorkspacePerDiemDetailsPage.tsx#L46C48-L46C63

Update to

```diff

    const [hasNavigatedBack, setHasNavigatedBack] = useState(false);
    let FullPageBlockingView;

    if (hasNavigatedBack) {
        FullPageBlockingView = View;
    } else if (isEmptyObject(selectedSubRate)) {
        FullPageBlockingView = FullPageOfflineBlockingView;
    } else {
        FullPageBlockingView = View;
    }
```


https://github.com/Expensify/App/blob/0ee0a123dac5b49f00da6c90619c03d5dabdd71f/src/pages/workspace/perDiem/WorkspacePerDiemDetailsPage.tsx#L48

Update to

```diff
    const handleDeletePerDiemRate = () => {
        ...
        setHasNavigatedBack(true);
        ...
    };
```

### What specific scenarios should we cover in automated tests to prevent reintroducing this issue in the future?


This is a UI bug and does not require a unit test

### What alternative solutions did you explore? (Optional)

**Reminder:** Please use plain English, be brief and avoid jargon. Feel free to use images, charts or pseudo-code if necessary. Do not post large multi-line diffs or write walls of text. Do not create PRs unless you have been hired for this job.

**Comment by Gonals** [comment]
This'll be fixed by @shubham1206agra as part of the per diem project
